mllib/src/test/java/org/apache/spark/mllib/clustering/JavaLDASuite.java

```diff
@@ -160,11 +160,42 @@ public void OnlineOptimizerCompatibility() {
     assertEquals(roundedLocalTopicSummary.length, k);
   }
 
+  @Test
+  public void logPerplexity(){
+    // Blindly copied from LDASuite.
+    JavaRDD<Tuple2<Long, Vector>> docs = sc.parallelize(toyData, 2);
+    assertEquals(toyModel.logPerplexity(JavaPairRDD.fromJavaRDD(docs)), 3.690, 1e-3);
```

Comment:
I would not test the value.  Just make sure it runs in Java.  Same for other tests.  Since it's difficult to test correctness for many ML algorithms, I like isolating correctness tests to Scala.

That should also make the below tests shorter.